airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/AirbyteDestination.java

```diff
@@ -4,9 +4,78 @@
 
 package io.airbyte.workers.protocols.airbyte;
 
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.config.WorkerDestinationConfig;
 import io.airbyte.protocol.models.AirbyteMessage;
-import io.airbyte.workers.protocols.Destination;
+import java.nio.file.Path;
+import java.util.Optional;
 
-public interface AirbyteDestination extends Destination<AirbyteMessage> {
+/**
+ * This interface provides a java interface over all interactions with a Destination from the POV of
+ * the platform. It encapsulates the full lifecycle of the Destination as well as any inputs and
+ * outputs.
+ */
+public interface AirbyteDestination extends CheckedConsumer<AirbyteMessage, Exception>, AutoCloseable {
+
+  /**
+   * Starts the Destination container. It instantiates a writer to write to STDIN on that container.
+   * It also instantiates a reader to listen on STDOUT.
+   *
+   * @param destinationConfig - contains the arguments that must be passed to the read method of the
+   *        Destination.
+   * @param jobRoot - directory where the job can write data.
+   * @throws Exception - throws if there is any failure in startup.
+   */
+  void start(WorkerDestinationConfig destinationConfig, Path jobRoot) throws Exception;
+
+  /**
+   * Accepts an AirbyteMessage and writes it to STDIN of the Destination. Blocks if STDIN's buffer is
+   * full.
+   *
+   * @param message message to send to destination.
+   * @throws Exception - throws if there is any failure in writing to Destination.
+   */
+  @Override
+  void accept(AirbyteMessage message) throws Exception;
+
+  /**
+   * This method is a flush to make sure all data that should be written to the Destination is
+   * written. (Potentially should just rename it to flush)
+   *
+   * @throws Exception - throws if there is any failure when flushing.
+   */
+  void notifyEndOfStream() throws Exception;
```

Comment:
It looks like this method gets called when [a destination is closed ](https://github.com/airbytehq/airbyte/blob/c99335ff8baf7642dbe06595308455b0206f7ef3/airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/DefaultAirbyteDestination.java#L106). Is it possible for this to happen when the buffer contains a very large amount of data (i.e. if the source is much faster than the destination), and if so wouldn't flushing the entire buffer to STDIN cause issues in the destination by flooding STDIN?

I guess I'm trying to understand how dangerous it could be to call this at any point in time.

Reply:
good call out. updated it to explain it better. does it help?

> This method is a flush to make sure all data that should be written to the Destination is written. Any messages that have already been accepted ({@link AirbyteDestination#accept(AirbyteMessage)} ()}) will be flushed. Any additional messages sent to accept will not be flushed. In fact, flush should fail if the caller attempts to send it additional messages after calling this method.